uaclient/jobs/gcp_auto_attach.py

```diff
@@ -8,27 +8,28 @@
 from uaclient.clouds.identity import get_cloud_type
 
 
-def gcp_auto_attach(cfg: config.UAConfig) -> None:
+def gcp_auto_attach(cfg: config.UAConfig) -> bool:
     # If the instance is already attached we will not do anything.
     # This implies that the user may have a new license attached to the
     # instance, but we will not perfom the change through this job.
     if cfg.is_attached:
-        return
+        return False
 
     # We will not do anything in a non-GCP cloud
     cloud_id, _ = get_cloud_type()
     if not cloud_id or cloud_id != "gce":
-        return
+        return False
 
     try:
         # This function already uses the assert lock decorator,
         # which means that we don't need to make create another
         # lock only for the job
         action_auto_attach(args=None, cfg=cfg)
```

Comment:
I think we can just `return action_auto_attach(args=None, cfg=cfg)`

Reply:
`action_auto_attach` will return 0 on success and 1 on failure, so it's the opposite of what the implementation expects. I preferred to put `True | False` instead of `something | None` to make clear that we are interested if the function actually executed something or not.

Which leads to a question: do we need to log every time the function tries to auto_attach, or should we watch for its return code to decide logging or not?

The attach process overall will produce logs anyway, so I think it should return `True` here in each try, so we get that the try to attach is timer related when looking at the logs.

Reply:
Sorry, I think I missing your point here. After a successful auto-attach this job should not be executed again because of the `cfg.is_attached` logic. Only if the user manually detaches that we will have to run it again. So why can't we just return the output of `action_auto_attach` ?

Reply:
Ohhh fair enough, my mistake here :facepalm: